keps/sig-node/4622-topologymanager-max-allowable-numa-nodes/README.md

```diff
@@ -0,0 +1,746 @@
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+In this KEP, we propose a new TopologyManager Policy Option called `max-allowable-numa-nodes` to configure the value of maxAllowableNUMANodes in the TopologyManager. the current hard-coded value of 8 was added as a stop-gap 4 years ago, a configurable option policy option can improve the topology manager to remove the state explosion that occurs when trying to enumerate the possible NUMA affinities and generating their hints. and should be sufficient to allow users to increase the limit when appropriate.
+
+what's the maxAllowableNUMANodes? maxAllowableNUMANodes specifies the maximum number of NUMA Nodes that the TopologyManager supports on the underlying machine. At present, having more than this number of NUMA Nodes will result in a state explosion when trying to enumerate possible NUMAAffinity masks and generate hints for them. As such, if more NUMA Nodes than this are present on a machine and the TopologyManager is enabled, an error will be returned and the TopologyManager will not be loaded.
+
+## Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Introduce a new TopologyManager Policy Option called `max-allowable-numa-nodes`.
```

Comment:
let's try to express the intent more clearly:

Support high-end CPUs with more than 8 NUMA cells.

Reply:
thanks! done.